vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java

```diff
@@ -65,14 +68,28 @@ public MailClient sendMail(MailMessage message, Handler<AsyncResult<MailResult>>
     Context context = vertx.getOrCreateContext();
     if (!closed) {
       if (validateHeaders(message, resultHandler, context)) {
-        connectionPool.getConnection(result -> {
-          if (result.succeeded()) {
-            result.result().setErrorHandler(th -> handleError(th, resultHandler, context));
-            sendMessage(message, result.result(), resultHandler, context);
-          } else {
-            handleError(result.cause(), resultHandler, context);
-          }
-        });
+        vertx.executeBlocking(
+            fut -> {
+              if (hostname == null) {
```

Comment:
(I'm pretty sure to have written that somewhere, but can't find it, so ignore it is it's a duplicate).

the hostname field is accessed from two threads: the event loop and the worker thread. The worker thread should not access it but compute the value and return it to the event loop thread who write and read it)